### PR TITLE
[MPS] Raise error for int64 inputs of dot operator.

### PR DESCRIPTION
**Pull request by kulinseth**

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/Blas.mm [hunk 1]
```diff
@@ -20,6 +20,9 @@ Tensor dot_mps(
   const Tensor &self,
   const Tensor &other)
 {
+
+  TORCH_CHECK(self.scalar_type() != ScalarType::Long, "MPS: dot op doesn't support int64 input")
```
Comment (DenisVieriu97) [inline aten/src/ATen/native/mps/operations/Blas.mm:24]: missing semicolon
```suggestion
  TORCH_CHECK(self.scalar_type() != ScalarType::Long, "MPS: dot op doesn't support int64 input");
```

Comment (kulinseth) [reply aten/src/ATen/native/mps/operations/Blas.mm:24]: You don't need to have semicolon for this.

**Review by razarmehr — APPROVED** [review]
LGTM

**Pull request closed without merging**